File: nodejs/src/state/PreparingState.ts
import OrderState, { IOrder } from "./OrderState";
import ReadyState from "./ReadyState";
import CancelledState from "./CancelledState";

class PreparingState extends OrderState {
    constructor(order: IOrder) { super(order); }

    getName(): string { return "PREPARING"; }

    next(): void {
        console.log(`🍵 Đơn #${this.order.id}: PREPARING → READY`);
        this.order.setState(new ReadyState(this.order));
    }

    cancel(): void {
        console.log(`❌ Đơn #${this.order.id}: PREPARING → CANCELLED`);
        this.order.setState(new CancelledState(this.order));
    }
}

export default PreparingState;

File: spring/demo/src/main/java/com/example/demo/state/PreparingState.java
package com.example.demo.state;

import com.example.demo.model.Order;

public class PreparingState extends OrderState {
    public PreparingState(Order order) {
        super(order);
    }

    @Override
    public String getName() {
        return "PREPARING";
    }

    @Override
    public void next() {
        System.out.println("Đơn #" + order.getId() + ": PREPARING -> READY");
        order.setState(new ReadyState(order));
    }

    @Override
    public void cancel() {
        System.out.println("Đơn #" + order.getId() + ": PREPARING -> CANCELLED");
        order.setState(new CancelledState(order));
    }
}
